sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -257,59 +330,153 @@ bool SCIStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest& req, cav
+    ROS_DEBUG_STREAM("current state dtd: " << current_state.downtrack);
+    ROS_DEBUG_STREAM("stop_intersection_down_track dtd: " << stop_intersection_down_track);
+
+    resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
+          current_state.downtrack, stop_intersection_down_track, current_state.speed, current_state.lane_id,
+          current_state.lane_id, stop_acc, current_state.stamp,
+          current_state.stamp + ros::Duration(stop_duration)));
 
   }
-  else
+
+  bool time_for_crossing = (int(street_msg_timestamp_) - int(scheduled_enter_time_)) >= 0;
+  // time to cross intersection
+
+  ROS_DEBUG_STREAM("time for crossing? " << time_for_crossing);
+
+  if (time_for_crossing && is_allowed_int_)
   {
-    // Passed the stop line
-    // Intersection transit maneuver
+    ROS_DEBUG_STREAM("street_msg_timestamp_ - scheduled_enter_time_ = " << street_msg_timestamp_ - scheduled_enter_time_);
+
 
+    ROS_DEBUG_STREAM("Vehicle is crossing the intersection");
+    // Passing the stop line
     
     // Compose intersection transit maneuver
-    double intersection_transit_time = (scheduled_depart_time_ - scheduled_enter_time_)*1000.0;
-
-    double intersection_end_downtrack =
-      wm_->routeTrackPos(nearest_stop_intersection->lanelets().back().centerline2d().back()).downtrack;
+    base_time = std::max(scheduled_enter_time_, street_msg_timestamp_);
+    double intersection_transit_time = (scheduled_depart_time_ - base_time)/1000;
 
     // Identify the lanelets which will be crossed by approach maneuvers lane follow maneuver
     std::vector<lanelet::ConstLanelet> crossed_lanelets =
           getLaneletsBetweenWithException(current_downtrack_, intersection_end_downtrack, true, true);
 
+    
+    // find the turn direction at intersection:
+
+    intersection_turn_direction_ = getTurnDirectionatIntersection(crossed_lanelets);
+    
+    ROS_DEBUG_STREAM("turn direction at the intersection is: " << intersection_turn_direction_);
+
     double intersection_speed_limit = findSpeedLimit(nearest_stop_intersection->lanelets().front());
 
     resp.new_plan.maneuvers.push_back(composeIntersectionTransitMessage(
-      current_downtrack_, intersection_end_downtrack, current_state.speed, intersection_speed_limit,
-      current_state.stamp, req.header.stamp + ros::Duration(intersection_transit_time), crossed_lanelets.front().id(), crossed_lanelets.back().id()));
+      current_state.downtrack, intersection_end_downtrack+20, current_state.speed, intersection_speed_limit,
+      current_state.stamp, req.header.stamp + ros::Duration(intersection_transit_time), intersection_turn_direction_, crossed_lanelets.front().id(), crossed_lanelets.back().id()));
+    
     // when passing intersection, set the flag to false
-    // TODO: Another option, check the time on mobilityoperation
-    approaching_stop_controlled_interction_ = false;
+    // TODO: Another option, check the time on mobilityoperation to see if it passes intersection
```

Comment:
Is this todo tracked somewhere as a github issue or JIRA story?

Reply:
Not needed. It was just an idea I had during the implementation